homeassistant/components/deluge/coordinator.py

```diff
@@ -14,11 +14,30 @@
 from homeassistant.exceptions import ConfigEntryAuthFailed
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
-from .const import LOGGER, DelugeGetSessionStatusKeys
+from .const import LOGGER, DelugeGetSessionStatusKeys, DelugeSensorType
 
 type DelugeConfigEntry = ConfigEntry[DelugeDataUpdateCoordinator]
 
 
+def count_states(data: dict[str, Any]) -> dict[str, int]:
+    """Count the states of the provided torrents."""
```

Comment:
someone else's example, but since I see all types of compression in review, you could do something like

```
from collections import Counter
(...)
    counts = Counter(torrent[b"state"].decode() for torrent in data.values())

    return {
        DelugeSensorType.DOWNLOADING_COUNT_SENSOR.value: counts.get("Downloading", 0),
        DelugeSensorType.SEEDING_COUNT_SENSOR.value: counts.get("Seeding", 0),
    }
```